src/models/app_data_json_format.rs

```diff
@@ -175,88 +174,69 @@ mod tests {
                         .unwrap(),
                     version: "6.6.6".to_string(),
                 }),
-                quote: Some(Quote::V2(QuoteV2 {
-                    version: String::from("2.0"),
-                    slippage_bips: 5,
-                })),
-            }),
-        };
-
-        assert_eq!(json, expected);
-    }
-    #[test]
-    fn test_loading_json_v4_with_amounts() {
-        let value = json!({
-                "version":"1.2.3",
-                "appCode":"MooSwap",
-                "environment": "production",
-                "metadata":{
-                    "environment": "production",
-                    "referrer":{
-                        "kind":"referrer",
-                        "address":"0x8c35B7eE520277D14af5F6098835A584C337311b",
-                        "version":"6.6.6"
-                },
-                "quote": {
-                    "version": "2.0",
-                    "buyAmount": "123456789",
-                    "sellAmount": "123456789101112",
-                }
-            }
-        });
-        let json: AppData = serde_json::from_value(value).unwrap();
-        let expected = AppData {
-            version: "1.2.3".to_string(),
-            app_code: "MooSwap".to_string(),
-            environment: Some("production".to_string()),
-            metadata: Some(Metadata {
-                environment: Some("production".to_string()),
-                referrer: Some(Referrer {
-                    address: "0x8c35B7eE520277D14af5F6098835A584C337311b"
-                        .parse()
-                        .unwrap(),
-                    version: "6.6.6".to_string(),
-                }),
-                // Notice that although we have specified v2 for quote, we fall back on the v1 quote.
-                // This is much cleaner than trying to implement optional fields for each of
-                // the three components and validate that
-                // only one of slippage OR (both of sell/buy amounts are populated)
-                quote: Some(Quote::V1(QuoteV1 {
-                    version: String::from("2.0"),
-                    buy_amount: U256::from_dec_str("123456789").unwrap(),
-                    sell_amount: U256::from_dec_str("123456789101112").unwrap(),
-                })),
+                quote: Some(Quote::V2(QuoteV2 { slippage_bips: 5 })),
             }),
         };
 
         assert_eq!(json, expected);
     }
 
     #[test]
-    fn test_loading_quote() {
+    fn test_loading_quote_ok() {
         let v1_value = json!({
-            "version": "2.0",
+            "version": "0.1.0",
             "sellAmount": "123",
             "buyAmount": "4567",
```

Comment:
That is interesting to know. I wasn't aware since it wasn't part of this code base before and this is only about v2. Maybe @josojo has more insight as to what we should do about this.